Revert "[tools] Add option for device specific image creation tools"

kjokinie opened the pull request:
This reverts commit f6bf842884bd5dfa458dd8b2b868337fc2c78d6f.

If device specific tools are needed, one should just modify a
device specific droid-hal-tools package for their adaptation.
It is much quicker and works better with legacy functionality.

Signed-off-by: Kalle Jokiniemi <kalle.jokiniemi@jolla.com>

kjokinie commented:
Version 2 of the PR:

Now includes a new patch the builds the tools same way as the generic droid-hal-tools OBS package does. Please check the patch description for more details.

sledges commented on droid-hal-device.inc:
```diff
@@ -191,6 +197,29 @@ echo Running droid build in HABUILD_SDK
+      %{android_root}/system/core/libsparse/simg2img
+
+if test -f rpm/helpers/mkbootimg.mk; then
+    make -f `pwd`/rpm/helpers/mkbootimg.mk -C %{android_root}/system/core/mkbootimg/
```
`pwd` here and below, should not be needed, as build dir will always be $ANDROID_ROOT (where rpm/ resides)

kjokinie replied:
Hmm.. seems when using the -C option for make makes the current pwd change for duration of the make run, and then the line does not work without the pwd. Tried with mb2 building.

kjokinie replied:
In other words, you need to have absolute path for the .mk file there.

sledges commented on droid-hal-device.inc:
```diff
@@ -191,6 +197,29 @@ echo Running droid build in HABUILD_SDK
+if test -f rpm/helpers/mkbootimg.mk; then
+    make -f `pwd`/rpm/helpers/mkbootimg.mk -C %{android_root}/system/core/mkbootimg/
+else
+    make -f `pwd`/rpm/dhd/helpers/mkbootimg.mk -C %{android_root}/system/core/mkbootimg/
```
replace rpm/dhd with %{dhd_path}, here and below

saukko commented on droid-hal-device.inc:
```diff
@@ -102,6 +101,22 @@ Device specific droid headers for %{rpm_device}.
 
 ################
+%package tools
+Provides: droid-hal-tools
```
We do not need this as hw adaptation repo can make its own decision which package to provide and after this we do not need to have droid-hal-tools at all anymore.

kjokinie commented:
Thanks, good comments. I'll fix and update PR.

kjokinie commented:
Version 3:

* Removed `pwd` from test lines, as it's not needed there

* Replaced depracated `...` with $(...) calls

* Use %{dhd_path} istead of rpm/dhd/

sledges commented on droid-hal-device.inc:
```diff
@@ -191,6 +197,29 @@ echo Running droid build in HABUILD_SDK
+      %{android_root}/system/core/libsparse/simg2img
+
+if test -f rpm/helpers/mkbootimg.mk; then
+    make -f $(pwd)/rpm/helpers/mkbootimg.mk -C %{android_root}/system/core/mkbootimg/
```
We should keep it consistent with current "make" approach used here: https://github.com/mer-hybris/droid-hal-device/blob/f6bf842884bd5dfa458dd8b2b868337fc2c78d6f/droid-hal-device.inc#L205
It also needs testing whether this line (make -f $(pwd)/rpm... , or after fixing the bit above: (cd rpm/; make ...) ) will build on OBS wrt to correct paths. @lbt, I think your glance is enough, instead of full test build.

kjokinie commented:
Version 4:

* Moved the tools build into the normal /helpers/makefile

* Overrides checked at dhd.inc and passed over via <TOOLNAME>_MK environment variables to helpers makefile build

sledges commented:
LGTM
